loco.py

```diff
@@ -1,66 +1,98 @@
+#!/usr/bin/env python3
+
+import re
 import heapq
 import sys
 
 
-class LogFile(object):
+class Pushable:
```

Comment:
Yeah, if there's an existing library less code is always better. We don't need to worry about multiple continuation lines, though, as we'll only ever have to push 1. It's basically an implementation of peek.

Want me to knock that up?